app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -118,60 +119,68 @@ public AppViewPresenter(AppViewView view, AccountNavigator accountNavigator,
 
-  private Completable showBannerAd() {
-    return appViewManager.shouldLoadBannerAd()
-        .observeOn(viewScheduler)
-        .flatMapCompletable(loadBanner -> {
-          if (loadBanner) {
-            view.showBannerAd();
-          }
-          return Completable.complete();
-        });
+    loadInterstitialAd();
+    showInterstitialNew();
   }
 
-  private void showInterstitialAd() {
+  private void showInterstitialNew() {
     view.getLifecycleEvent()
         .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
-        .flatMap(created -> view.installAppClick())
-        .flatMap(create -> appViewManager.loadAppViewViewModel()
-            .toObservable())
+        .flatMap(__ -> view.isAppViewReadyToDownload())
+        .flatMap(__ -> Observable.zip(downloadStarted(), view.interstitialAdLoaded(),
+            (downloadAppViewModel, moPubInterstitialAdClickType) -> Observable.just(
+                downloadAppViewModel)))
+        .doOnNext(__ -> view.showInterstitialAd())
+        .doOnNext(__ -> appViewAnalytics.installInterstitialImpression())
+        .observeOn(Schedulers.io())
+        .flatMapSingle(__ -> appViewManager.recordInterstitialImpression())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, throwable -> crashReport.log(throwable));
+  }
+
+  private Observable<DownloadAppViewModel> downloadStarted() {
+    return appViewManager.loadAppViewViewModel()
+        .toObservable()
         .filter(app -> !app.isLoading())
         .filter(app -> !app.isAppCoinApp())
         .flatMap(app -> appViewManager.loadDownloadAppViewModel(app.getMd5(), app.getPackageName(),
             app.getVersionCode(), app.isPaid(), app.getPay())
             .filter(model -> model.getDownloadModel()
-                .isDownloading())
-            .first()
-            .observeOn(Schedulers.io())
-            .flatMapSingle(model -> appViewManager.shouldLoadInterstitialAd())
-            .filter(loadInterstitial -> loadInterstitial)
-            .observeOn(viewScheduler)
-            .doOnNext(__ -> view.initInterstitialAd())
-            .delay(1, TimeUnit.SECONDS)
-            .observeOn(viewScheduler)
-            .doOnNext(__ -> view.loadInterstitialAd()))
+                .isDownloading()
+                && model.getDownloadModel()
+                .getProgress() >= 5
+                && model.getDownloadModel()
+                .getProgress() < 100)
+            .first());
+  }
+
+  private void loadInterstitialAd() {
+    view.getLifecycleEvent()
+        .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
+        .flatMap(created -> view.isAppViewReadyToDownload())
+        .flatMapSingle(model -> appViewManager.shouldLoadInterstitialAd())
+        .filter(loadInterstitial -> loadInterstitial)
+        .observeOn(viewScheduler)
+        .doOnNext(__ -> view.initInterstitialAd())
+        .doOnNext(__ -> view.loadInterstitialAd())
```

Comment:
I would aggregate both methods into a single load method or something, since this might lead to confusion and it is an unnecessary separation.